analyzers/src/SonarAnalyzer.CFG/Roslyn/CfgAllPathValidator.cs

```diff
@@ -23,29 +23,47 @@ namespace SonarAnalyzer.CFG.Roslyn
     public abstract class CfgAllPathValidator
     {
         private readonly ControlFlowGraph cfg;
-        private readonly Dictionary<BasicBlock, bool> visitedStatus = new Dictionary<BasicBlock, bool>();
+        private readonly HashSet<BasicBlock> visitedStatus = [];
```

Comment:
This doesn't need to be a field anymore, so the `visited` name will work nicely as a local variable.